lib/VM/JSLib/RegExp.cpp

```diff
@@ -460,6 +461,65 @@ static void createGroupsObject(
   JSObject::setNamedSlotValueUnsafe(matchObj.get(), runtime, groupsDesc, shv);
 }
 
+static void createIndicesArray(
+    Runtime &runtime,
+    Handle<JSArray> matchObj,
+    Handle<JSObject> mappingObj,
+    Handle<JSArray> indices) {
+  // If there are no capture groups, then set groups to undefined.
+  if (!mappingObj) {
+    auto executionStatus = JSObject::defineNewOwnProperty(
```

Comment:
You should refrain from using `defineNewOwnProperty`. You can only use this API when you know for certain that the property you're adding does not exist on the current object. To be safe, use `defineOwnProperty`.